### PR TITLE
Add test cases and fix for StripMetadata: false

### PR DESCRIPTION
**Pull request by greut**
Test cases for #194 where `StripMetadata` seems to be applied when it goes into libjpeg shrink on load (`vipsShrinkJpeg`).

Signed-off-by: Yoan Blanc <yoan@dosimple.ch>

### CODE REVIEW

File: resizer.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func resizer(buf []byte, o Options) ([]byte, error) {
 	}
 
 	// Try to use libjpeg shrink-on-load
-	if imageType == JPEG && shrink >= 2 {
+	/*if imageType == JPEG && shrink >= 2 {
 		tmpImage, factor, err := shrinkJpegImage(buf, image, factor, shrink)
```
Comment (greut) [inline resizer.go]: `shrinkJpegImage` ~ `vips_jpegload_buffer` strips the metadata.

Comment (greut) [reply resizer.go]: @h2non should it avoid using it when `StripMetadata: false` ?

**Comment by Divitay** [comment]
@greut Tell me, please, how was the test? Fixed a problem?

**Comment by greut** [comment]
@Divitay if it goes through libjpeg _shrink on load_ feature, the metadata are stripped no matter what.

**Comment by Divitay** [comment]
Nothing can be done about this?
How to be?

**Comment by greut** [comment]
I dunno ping the libvips author, or the libjpeg-turbo ones.

File: resizer.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func resizer(buf []byte, o Options) ([]byte, error) {
 	}
 
 	// Try to use libjpeg shrink-on-load
-	if imageType == JPEG && shrink >= 2 {
+	if imageType == JPEG && shrink >= 2 && o.StripMetadata {
```
Comment (h2non) [inline resizer.go]: I'm afraid this is going to introduce behavioral breaking changes somehow. Did you test this in detail?

Comment (greut) [reply resizer.go]: I tried to understand where the metadata were stripped within libvips but didn't really succeed...

**Pull request closed without merging**